lib/code-ribbon-single-ribbon.js

```diff
@@ -50,11 +54,47 @@ class CodeRibbonSingleRibbon extends PaneAxis {
     };
   }
 
+  at_least_n_empty_right_patches(target_num = 2) {
+    if (target_num > this.children.length) return false;
```

Comment:
Is this because we only want 1 set of column patches being rendered off-screen?

Reply:
This is so that we can check if the last N patches on the right end of a ribbon are empty, and if some number of them are (e.x. a full screen of empty patches) then we can remove/trim them from the end.